File: Npc/11001868.cs
using Maple2.Model.Enum;
using Maple2.Script.Npc;

/// <summary>
/// 11001868: Rachael
/// </summary>
public class _11001868 : NpcScript {
    protected override int First() {
        return 20;
    }

    // Select 0:
    // $script:1213150207007549$
    // - Ah, nice to see you!
    protected override int Select() => 0;

    protected override int Execute(int selection) {
        switch (Id, Index++) {
            case (20, 0):
                // $script:1213150207007551$
                // - This mine is run by the Nerman Guild. We've got all the materials a blacksmith might need.
                return -1;
        }

        return default;
    }

    protected override NpcTalkButton Button() {
        return (Id, Index) switch {
            (20, 0) => NpcTalkButton.Close,
            _ => NpcTalkButton.None,
        };
    }
}

File: Npc/11001868.py
""" 11001868: Rachael """
from npc_api import Option, Script
import random


class Main(Script):
    def first(self) -> int:
        return 20

    def select(self) -> int:
        return 0

    def __0(self, pick: int) -> int:
        # $script:1213150207007549$
        # - Ah, nice to see you!
        return None # TODO

    def __20(self, pick: int) -> int:
        # $script:1213150207007551$
        # - This mine is run by the Nerman Guild. We've got all the materials a blacksmith might need.
        return -1

    def button(self) -> Option:
        if (self.state, self.index) == (20, 0):
            return Option.CLOSE
        return Option.NONE
